README.md

```diff
@@ -37,3 +37,60 @@ SpiNNaker.  Tools exist for:
  - Clock discipline
  - Routing table minimisation
  - Optimal routing key allocation
+
+## Developing Rig
+
+### Installing in a `virtualenv`
+
+Create a new [virtualenv](https://pypi.python.org/pypi/virtualenv) by running:
+
+    virtualenv new_directory_name
+
+Then activate the virtualenv before installing.
+
+    cd new_directory_name
+    . bin/activate
+
+Clone and install.
+
+    git clone git@github.com:mundya/rig
+    cd rig
+    python setup.py install develop
+
+This will install the requirements and provide you with a sandboxed environment
+for testing and working with rig.  To leave the virtualenv just run
+`deactivate`.  Run `. bin/activate` whenever you want to re-enter the
+environment.
+
+### Running the tests
+
+We use [py.test](http://pytest.org) to test rig,
+[pytest-cov](https://pypi.python.org/pypi/pytest-cov/1.8.1) to generate
+coverage reports and the [flake8](https://pypi.python.org/pypi/flake8) coding
+standard checker.
+
+Install py.test et al. in the virtualenv.
+
+    pip install pytest pytest-cov flake8
```

Comment:
Surely there a way to make setup tools install test stuff? (there's a `requirements-test.txt` in after all -- though I ended up doing this by hand myself since a quick google did not yield an answer...)